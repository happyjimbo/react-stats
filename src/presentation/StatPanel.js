import React, {PropTypes, Component} from 'react';
import {Panel} from 'react-bootstrap';

class StatPanel extends Component {

    render() {

        const display = this.props.display;
        const text = this.props.text;

        // Use SCSS instead of inline.
        const divStyle = {
            wordWrap: "break-word"
        };

        return (
            <div>
                <Panel collapsible expanded={display} style={divStyle}>
                    {text}
                </Panel>
            </div>
        );
    }

    static propTypes = {
        display: PropTypes.bool.isRequired,    
        text: PropTypes.string.isRequired
    }  
}

export default StatPanel;