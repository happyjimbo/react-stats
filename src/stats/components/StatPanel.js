import React, {PropTypes, Component} from 'react'
import {Panel} from 'react-bootstrap'
import LineGraphContainer from '../../shared/containers/LineGraphContainer'
import StatLineGraphSelector from '../selectors/StatLineGraphSelector'

const StatPanel = ({panelStats, statName, display}) => {

    const style = {
        border: 'none',
        margin: 'auto',
        width: "100%",
        marginBottom: '1em'
    }

    return (
        <div>
            <Panel collapsible expanded={display} style={style}>
                <LineGraphContainer items={panelStats} statName={statName} selector={StatLineGraphSelector} />
            </Panel>
        </div>
    )
}

StatPanel.PropTypes = {
    panelStats: PropTypes.array.isRequired,
    statName: PropTypes.string.isRequired,
    display: PropTypes.bool.isRequired
}  

export default StatPanel
