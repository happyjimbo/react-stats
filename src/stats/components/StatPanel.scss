.StatPanel {
    .panel {
        border: 0;
        margin : 'auto';
        width : '100%';
        margin-bottom: '1em';
    }
}