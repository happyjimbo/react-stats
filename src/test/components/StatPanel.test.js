import React from 'react';
import {shallow} from 'enzyme';
import StatPanel from '../../components/StatPanel';
import renderer from 'react-test-renderer';

describe('StatPanel', () => {

    const statName = "MyStatName";
    const text = "some text";
    

    function setUp(extraProps = undefined) {

        const defaultProps = {
            statName,
            text,
        };

        const props = Object.assign({}, defaultProps, extraProps);
        const component = shallow(<StatPanel {...props} />);

        return {
            props,
            component
        }
    }

    it('compares component to snapshot', () => {
        
        const component = renderer.create(
            <StatPanel text="hello world" display={true} />
        );

        let statPanel = component.toJSON();
        expect(statPanel).toMatchSnapshot();
    });

    it ('sets expanded to true', () => {

        const props = { display: true };
        const { component } = setUp(props);

        const componentProps = component.find('Panel').props();
        expect(componentProps.expanded).toBe(true);
    });

    it ('sets expanded to false', () => {

        const props = { display: false };
        const { component } = setUp(props);

        const componentProps = component.find('Panel').props();
        expect(componentProps.expanded).toBe(false);
    });

    it ('displays text', () => {
        
        const props = { display: true };
        const { component } = setUp(props);

        const containsStatName = component.find('Panel').children().contains(text);
        expect(containsStatName).toBe(true);
    });
});