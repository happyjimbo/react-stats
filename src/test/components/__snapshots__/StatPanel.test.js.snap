exports[`StatPanel compares component to snapshot 1`] = `
<div>
  <div
    className="panel panel-default"
    id={null}
    style={
      Object {
        "wordWrap": "break-word",
      }
    }>
    <div
      aria-expanded={null}
      aria-hidden={false}
      className="panel-collapse collapse in"
      id={undefined}
      role={undefined}>
      <div
        className="panel-body">
        hello world
      </div>
    </div>
  </div>
</div>
`;
